test/testnet/unit/ODSafeManager.t.sol

```diff
@@ -0,0 +1,457 @@
+
+    vm.expectEmit(true, true, false, true);
+
+    emit OpenSAFE(_scenario.user, _scenario.userProxy, 1);
+
+    vm.mockCall(address(vault721), abi.encodeWithSelector(IVault721.mint.selector), abi.encode());
+
+    _scenario.safeId = safeManager.openSAFE('i', _scenario.userProxy);
+
+    assertEq(_scenario.safeId, 1, 'incorrect safeId returned');
+  }
+
+  event AllowSAFE(address indexed _sender, uint256 indexed _safe, address _usr, bool _ok);
+
+  function test_AllowSafe(Scenario memory _scenario) public happyPath(_scenario) {
+    vm.startPrank(_scenario.userProxy);
+
+    vm.expectEmit();
+    emit AllowSAFE(_scenario.userProxy, _scenario.safeId, owner, true);
+
+    safeManager.allowSAFE(_scenario.safeId, owner, true);
+    uint256[] memory _safes = safeManager.getSafes(_scenario.userProxy);
+    assertEq(_safes.length, 1, 'incorrect number of safes');
+    assertEq(_safes[0], 1, 'incorrect safe id');
+  }
+
+  function test_AllowSafe_Revert_OnlySafeOwner(Scenario memory _scenario) public {
+    _openSafe(_scenario);
+    vm.expectRevert(IODSafeManager.OnlySafeOwner.selector);
+    safeManager.allowSAFE(_scenario.safeId, _scenario.userProxy, true);
+  }
+
+  event TransferSAFEOwnership(address indexed _sender, uint256 indexed _safe, address _dst);
+
+  function test_transferSAFEOwnership(Scenario memory _scenario) public happyPath(_scenario) {
+    vm.startPrank(address(vault721));
+    vm.expectEmit();
+
+    emit TransferSAFEOwnership(address(vault721), 1, address(_scenario.user));
+
+    safeManager.transferSAFEOwnership(1, address(_scenario.user));
+
+    uint256[] memory _safes = safeManager.getSafes(_scenario.user);
+    assertEq(_safes.length, 1, 'SAFE transfer: incorrect number of safes');
+    assertEq(_safes[0], 1, 'SAFE transfer: incorrect safe id');
+  }
+
+  function test_transferSAFEOwnership_Revert_Vault721() public {
+    vm.expectRevert('SafeMngr: Only Vault721');
+    safeManager.transferSAFEOwnership(1, address(0x6));
+  }
+
+  function test_transferSAFEOwnership_Revert_ZeroAddress() public {
+    vm.expectRevert(IODSafeManager.ZeroAddress.selector);
+    vm.prank(address(vault721));
+    safeManager.transferSAFEOwnership(1, address(0));
+  }
+
+  function test_transferSAFEOwnership_Revert_AlreadySafeOwner(Scenario memory _scenario) public {
+    _openSafe(_scenario);
+    vm.expectRevert(IODSafeManager.AlreadySafeOwner.selector);
+    vm.prank(address(vault721));
+    safeManager.transferSAFEOwnership(_scenario.safeId, _scenario.userProxy);
+  }
+
+  event AllowHandler(address indexed _sender, address _usr, bool _ok);
+
+  function test_AllowHandler(Scenario memory _scenario) public happyPath(_scenario) {
+    vm.startPrank(_scenario.userProxy);
+
+    vm.expectEmit();
+    emit AllowHandler(_scenario.userProxy, _scenario.rando, true);
+
+    safeManager.allowHandler(_scenario.rando, true);
+
+    assertTrue(safeManager.handlerCan(_scenario.userProxy, 0, _scenario.rando), 'handler not allowed');
+  }
```

Comment:
Add cases for:

1. allowHandler successfully updated with new user address
2. data.nonce should be updated when allowHandler is updated
3. data.nonce should be updated when the vault is transferred

Reply:
> 3.  data.nonce should be updated when the vault is transferred.
transferSAFEOwnership test now checks for an updated nonce.



 I'm not sure what you're requesting with #1 and #2.  
Do you want me to change the allowHandler function to update the safe handler in `_safeData[_safeId].safeHandler` ?  
also, as it is allowHandler does not update the vault nonce.


here is the current allowHandler function on line 137 of the ODSafeManager:

``` 
 function allowHandler(address _usr, bool _ok) external {
    address safeHandler = msg.sender;
    SAFEData memory data = getSafeDataFromHandler(safeHandler);
    handlerCan[safeHandler][data.nonce][_usr] = _ok;
    emit AllowHandler(safeHandler, _usr, _ok);
  }
```

Reply:
Ok, you can ignore 1 and 2. 

For 3, this should be added in the section for `transferSAFEOwnership`